src/main/ncdio_pio.F90.in

```diff
@@ -1752,7 +1748,11 @@ contains
 #elif ({ITYPE}==TYPEINT)
        call pio_write_darray(ncid, vardesc, iodesc_plus%iodesc, data, status, fillval=ispval)
 #elif ({ITYPE}==TYPEDOUBLE)
-       call pio_write_darray(ncid, vardesc, iodesc_plus%iodesc, data, status, fillval=spval)
+       if (iodesc_plus%type == pio_double) then
+          call pio_write_darray(ncid, vardesc, iodesc_plus%iodesc, data, status, fillval=spval)
+       else
+          call pio_write_darray(ncid, vardesc, iodesc_plus%iodesc, real(data,kind=r4), status, fillval=real(spval,kind=r4))
```

Comment:
Doing the type conversion here rather than in pio reduces the communication requirement by 50%

Reply:
Thanks for the explanation. Does this also explain why you added some pio_inq_vartype calls? i.e., are you getting the variable type on the file rather than using the variable type passed into the routine?

And just to double-check: will these changes work with both pio1 and pio2?